pkg/apiclient/argo-kube-client.go

```diff
@@ -53,15 +53,15 @@ func newArgoKubeClient(clientConfig clientcmd.ClientConfig, instanceIDService in
 }
 
 func (a *argoKubeClient) NewWorkflowServiceClient() workflowpkg.WorkflowServiceClient {
-	return &argoKubeWorkflowServiceClient{workflowserver.NewWorkflowServer(a.instanceIDService, argoKubeOffloadNodeStatusRepo)}
+	return &errorTranslatingWorkflowServiceClient{&argoKubeWorkflowServiceClient{workflowserver.NewWorkflowServer(a.instanceIDService, argoKubeOffloadNodeStatusRepo)}}
```

Comment:
Is one more wrapper need for just translating errors?

Reply:
correct

Reply:
Do we need this wrapper?  Can you do it in an existing function?  So we can avoid the few files and complexity. Is there any advantage of using this way?

```
func (c argoKubeWorkflowServiceClient) CreateWorkflow(ctx context.Context, req *workflowpkg.WorkflowCreateRequest, _ ...grpc.CallOption) (*v1alpha1.Workflow, error) {
	resp, err := c.delegate.CreateWorkflow(ctx, req)
	if err != nil {
		return nil, grpcutil.TranslateError(err)
	}
	return resp, err
}
```

Reply:
I could combine the code into existing files, but it would violate SRP and result in future mistakes where people fail to implement the code